### ISSUE TITLE
Display a nicer error when virtualenv/pip aren't installed

### ISSUE THREAD

**Issue by jdm** [servo/servo#7630]
If you run a mach command and are missing the `virtualenv` package, the error looks like this:
```
Traceback (most recent call last):
  File "./mach", line 25, in <module>
    main(sys.argv)
  File "./mach", line 21, in main
    mach = mach_bootstrap.bootstrap(topdir)
  File "./python/mach_bootstrap.py", line 118, in bootstrap
    _activate_virtualenv(topdir)
  File "./python/mach_bootstrap.py", line 87, in _activate_virtualenv
    subprocess.check_call([virtualenv, "-p", python, virtualenv_path])
  File "/usr/lib64/python2.7/subprocess.py", line 535, in check_call
    retcode = call(*popenargs, **kwargs)
  File "/usr/lib64/python2.7/subprocess.py", line 522, in call
    return Popen(*popenargs, **kwargs).wait()
  File "/usr/lib64/python2.7/subprocess.py", line 710, in __init__
    errread, errwrite)
  File "/usr/lib64/python2.7/subprocess.py", line 1335, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory
```
We can do better if we catch the exception in mach_bootstrap.py when using `subprocess.check_call`.

Code: python/mach_bootstrap.py

**Comment by AnthonyBroadCrawford** [comment]
If you guys don't mind.  I'm going to take a crack at this.  I don't see this in the guidelines, but is there a naming convention to branches you guys recommend?

**Comment by jdm** [comment]
Branch names are entirely up to you, since they don't affect other developers at all.

**Comment by AnthonyBroadCrawford** [comment]
Awesome.  Thank you.

**Comment by AnthonyBroadCrawford** [comment]
Should have a PR sometime tomorrow.  Really enjoyed getting the lay of the land, albeit only in the small section I'm in.

**Comment by AnthonyBroadCrawford** [comment]
PR for this issue has been created (https://github.com/servo/servo/pull/7678).

Given this is my first Servo PR please provide any/all feedback, I'd appreciate it.

**Comment by frewsxcv** [comment]
Fixed in #7678 by @AnthonyBroadCrawford 

Thanks again!
